lms/djangoapps/courseware/courses.py

```diff
@@ -376,15 +376,15 @@ def get_courses(user, org=None, filter_=None):
     Returns a list of courses available, sorted by course.number and optionally
     filtered by org code (case-insensitive).
     """
+    include_hidden = filter_.pop('include_hidden', False) if filter_ else False
     courses = branding.get_visible_courses(org=org, filter_=filter_)
 
     permission_name = configuration_helpers.get_value(
         'COURSE_CATALOG_VISIBILITY_PERMISSION',
         settings.COURSE_CATALOG_VISIBILITY_PERMISSION
     )
 
-    courses = [c for c in courses if has_access(user, permission_name, c)]
-
+    courses = [c for c in courses if include_hidden or has_access(user, permission_name, c)]
```

Comment:
@douglashall good point, I think I can specify that my change should only apply for the see_in_catalog permission?  
I think effectively what I'm doing is setting the catalog visibility property to true.
`    def can_see_in_catalog():
        """
        Implements the "can see course in catalog" logic if a course should be visible in the main course catalog
        In this case we use the catalog_visibility property on the course descriptor
        but also allow course staff to see this.
        """
        return (
            _has_catalog_visibility(courselike, CATALOG_VISIBILITY_CATALOG_AND_ABOUT)
            or _has_staff_access_to_descriptor(user, courselike, courselike.id)
        )`
Since this is an or, this particular permission will always be true if I set include_hidden to true
Does that make sense?